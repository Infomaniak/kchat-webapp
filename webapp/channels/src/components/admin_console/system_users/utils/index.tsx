// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {SortingState} from '@tanstack/react-table';
import React from 'react';
import {FormattedMessage} from 'react-intl';

import {UserReportSortColumns, ReportSortDirection} from '@mattermost/types/reports';
import type {UserReportOptions, UserReport} from '@mattermost/types/reports';
import type {Team} from '@mattermost/types/teams';

import {PAGE_SIZES} from 'components/admin_console/list_table';

import type {AdminConsoleUserManagementTableProperties} from 'types/store/views';

import {ColumnNames, RoleFilters, StatusFilter, TeamFilters} from '../constants';
import type {TableOptions} from '../system_users';
import type {OptionType as TeamFilterOptionType} from '../system_users_filters_popover/system_users_filter_team';

export function convertTableOptionsToUserReportOptions(tableOptions?: TableOptions): UserReportOptions {
    return {
        page_size: tableOptions?.pageSize || PAGE_SIZES[0],
        from_column_value: tableOptions?.fromColumnValue,
        from_id: tableOptions?.fromId,
        direction: tableOptions?.direction,
        ...getSortColumnForOptions(tableOptions?.sortColumn),
        ...getSortDirectionForOptions(tableOptions?.sortIsDescending),
        ...getSearchFilterOption(tableOptions?.searchTerm),
        ...getTeamFilterOption(tableOptions?.filterTeam),
        ...getStatusFilterOption(tableOptions?.filterStatus),
        ...getRoleFilterOption(tableOptions?.filterRole),
        date_range: tableOptions?.dateRange,
    };
}

/**
 * Converts the sorting column name to API compatible sorting column name. Default sorting column name is by username.
 */
export function getSortColumnForOptions(id?: SortingState[0]['id']): Pick<UserReportOptions, 'sort_column'> {
    let sortColumn: UserReportOptions['sort_column'];

    if (id === ColumnNames.email) {
        sortColumn = UserReportSortColumns.email;
    } else if (id === ColumnNames.createAt) {
        sortColumn = UserReportSortColumns.createAt;
    } else {
        // Default sorting to first User details column
        sortColumn = UserReportSortColumns.username;
    }

    return {
        sort_column: sortColumn,
    };
}

/**
 * Converts the sorting direction to API compatible sorting direction. Default sorting direction is ascending.
 */
export function getSortDirectionForOptions(desc?: SortingState[0]['desc']): Pick<UserReportOptions, 'sort_direction'> {
    let sortDirection: UserReportOptions['sort_direction'];

    if (desc) {
        sortDirection = ReportSortDirection.descending;
    } else {
        sortDirection = ReportSortDirection.ascending;
    }

    return {
        sort_direction: sortDirection,
    };
}

/**
 * It returns the value of that column on which sorting is applied.
 */
export function getSortableColumnValueBySortColumn(row: UserReport, sortColumn: AdminConsoleUserManagementTableProperties['sortColumn']): string {
    switch (sortColumn) {
    case ColumnNames.email:
        return row.email;
    case ColumnNames.createAt:
        return String(row.create_at);
    default:
        return row.username;
    }
}

export function getTeamFilterOption(teamId?: string): Partial<Pick<UserReportOptions, 'team_filter' | 'has_no_team'>> {
    if (!teamId || teamId === TeamFilters.AllTeams) {
        return {
            team_filter: undefined,
            has_no_team: undefined,
        };
    } else if (teamId === TeamFilters.NoTeams) {
        return {
            team_filter: undefined,
            has_no_team: true,
        };
    }

    return {
        team_filter: teamId,
        has_no_team: undefined,
    };
}

export function getStatusFilterOption(status?: string): Partial<Pick<UserReportOptions, 'hide_active' | 'hide_inactive'>> {
    if (status === StatusFilter.Active) {
        return {
            hide_inactive: true,
        };
    } else if (status === StatusFilter.Deactivated) {
        return {
            hide_active: true,
        };
    }

    return {
        hide_active: undefined,
        hide_inactive: undefined,
    };
}

export function getPaginationInfo(pageIndex: number, pageSize: number, currentLength: number, total?: number) {
    if (!currentLength) {
        return (
            <FormattedMessage
                id='admin.system_users_list.pagination.no_users'
                defaultMessage='0 users'
            />
        );
    }

    const firstPage = (pageIndex * pageSize) + 1;
    const lastPage = (pageIndex * pageSize) + currentLength;
    const totalItems = total || 0;

    return (
        <FormattedMessage
            id='admin.system_users_list.pagination'
            defaultMessage='Showing {firstPage} - {lastPage} of {totalItems} users'
            values={{
                firstPage,
                lastPage,
                totalItems,
            }}
        />
    );
}

export function getDefaultSelectedValueFromList<T extends {value: string}>(value: string, options: T[]) {
    const option = options.find((option) => option.value === value);

    if (option) {
        return option;
    }

    return options[0];
}

export function getDefaultSelectedTeam(teamId: Team['id'] | string, label?: string): TeamFilterOptionType {
    if (!teamId || teamId === TeamFilters.AllTeams) {
        return {
            value: TeamFilters.AllTeams,
            label: (
                <FormattedMessage
                    id='admin.system_users.filters.team.allTeams'
                    defaultMessage='All Teams'
                />
            ),
        };
    } else if (teamId === TeamFilters.NoTeams) {
        return {
            value: TeamFilters.NoTeams,
            label: (
                <FormattedMessage
                    id='admin.system_users.filters.team.noTeams'
                    defaultMessage='No Teams'
                />
            ),
        };
    }

    return {
        value: teamId,
        label: label || '',
    };
}

export function getRoleFilterOption(role?: string): Pick<UserReportOptions, 'role_filter'> {
    if (!role || role === RoleFilters.Any) {
        return {role_filter: undefined};
    }
    return {role_filter: role};
}

export function getSearchFilterOption(search?: string): Pick<UserReportOptions, 'search_term'> {
    if (!search || search.trim().length === 0) {
        return {search_term: undefined};
    }

    return {search_term: search};
}
