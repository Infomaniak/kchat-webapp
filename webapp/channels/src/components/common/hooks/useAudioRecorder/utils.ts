export const getWasmFileURL = () =>
    new URL('wasm-media-encoders/wasm/mp3', import.meta.url);
