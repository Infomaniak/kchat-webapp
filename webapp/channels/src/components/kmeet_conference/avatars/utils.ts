export function countMeta<T>(
    items: T[],
    total = items.length,
): [T[], T[], {overflowUnnamedCount: number; nonDisplayCount: number}] {
    const breakAt = Math.max(items.length, total) > 4 ? 3 : 4;

    const displayItems = items.slice(0, breakAt);
    const overflowItems = items.slice(breakAt);

    const overflowUnnamedCount = Math.max(total - displayItems.length - overflowItems.length, 0);
    const nonDisplayCount = overflowItems.length + overflowUnnamedCount;

    return [displayItems, overflowItems, {overflowUnnamedCount, nonDisplayCount}];
}
