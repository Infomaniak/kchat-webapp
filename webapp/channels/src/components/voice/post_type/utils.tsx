import type {TranscriptData} from '@mattermost/types/transcript';

export function isValidTranscript(
    transcript: TranscriptData | null | undefined,
): transcript is TranscriptData & { text: string } {
    return (
        typeof transcript === 'object' &&
    transcript !== null &&
    typeof transcript.text === 'string' &&
    transcript.text.length > 0
    );
}
