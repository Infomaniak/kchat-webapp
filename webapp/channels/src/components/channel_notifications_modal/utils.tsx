// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage, defineMessages} from 'react-intl';

import type {ChannelNotifyProps} from '@mattermost/types/channels';
import type {UserNotifyProps} from '@mattermost/types/users';

import type {FieldsetCheckbox} from 'components/widgets/modals/components/checkbox_setting_item';
import type {FieldsetRadio} from 'components/widgets/modals/components/radio_setting_item';

import {NotificationLevels} from 'utils/constants';

export type ChannelMemberNotifyProps = Partial<ChannelNotifyProps> & Pick<UserNotifyProps, 'desktop_threads' | 'push_threads'>

const translations = defineMessages({
    MuteAndIgnoreSectionTitle: {
        id: 'channel_notifications.muteAndIgnore',
        defaultMessage: 'Mute or ignore',
    },
    NotifyMeTitle: {
        id: 'channel_notifications.NotifyMeTitle',
        defaultMessage: 'Notify me about…',
    },
    ThreadsReplyTitle: {
        id: 'channel_notifications.ThreadsReplyTitle',
        defaultMessage: 'Thread reply notifications',
    },

    DesktopNotificationsSectionTitle: {
        id: 'channel_notifications.desktopNotificationsTitle',
        defaultMessage: 'Desktop Notifications',
    },

    DesktopNotificationsSectionDesc: {
        id: 'channel_notifications.desktopNotificationsDesc',
        defaultMessage: 'Available on Chrome, Edge, Firefox, and the Mattermost Desktop App.',
    },

    MobileNotificationsSectionTitle: {
        id: 'channel_notifications.mobileNotificationsTitle',
        defaultMessage: 'Mobile Notifications',
    },

    MobileNotificationsSectionDesc: {
        id: 'channel_notifications.mobileNotificationsDesc',
        defaultMessage: 'Notification alerts are pushed to your mobile device when there is activity in Mattermost.',
    },

    MuteChannelDesc: {
        id: 'channel_notifications.muteChannelDesc',
        defaultMessage: 'Turns off notifications for this channel. You’ll still see badges if you’re mentioned.',
    },

    IgnoreMentionsDesc: {
        id: 'channel_notifications.ignoreMentionsDesc',
        defaultMessage: 'When enabled, @channel, @here and @all will not trigger mentions or mention notifications in this channel',
    },

    MuteChannelInputFieldTitle: {
        id: 'channel_notifications.muteChannelTitle',
        defaultMessage: 'Mute channel',
    },

    DesktopReplyThreadsInputFieldTitle: {
        id: 'channel_notifications.checkbox.threadsReplyTitle',
        defaultMessage: 'Notify me about replies to threads I\'m following',
    },

    MobileReplyThreadsInputFieldTitle: {
        id: 'channel_notifications.checkbox.threadsReplyTitle',
        defaultMessage: 'Notify me about replies to threads I\'m following',
    },

    sameMobileSettingsDesktopInputFieldTitle: {
        id: 'channel_notifications.checkbox.sameMobileSettingsDesktop',
        defaultMessage: 'Use the same notification settings as desktop',
    },

    IgnoreMentionsInputFieldTitle: {
        id: 'channel_notifications.ignoreMentionsTitle',
        defaultMessage: 'Ignore mentions for @channel, @here and @all',
    },

    AutoFollowThreadsTitle: {
        id: 'channel_notifications.autoFollowThreadsTitle',
        defaultMessage: 'Follow all threads in this channel',
    },

    AutoFollowThreadsDesc: {
        id: 'channel_notifications.autoFollowThreadsDesc',
        defaultMessage: 'When enabled, all new replies in this channel will be automatically followed and will appear in your Threads view.',
    },

    AutoFollowThreadsInputFieldTitle: {
        id: 'channel_notifications.checkbox.autoFollowThreadsTitle',
        defaultMessage: 'Automatically follow threads in this channel',
    },
});

const desktopNotificationInputFieldOptions = defineMessages({
    allNewMessages: {
        id: 'channel_notifications.desktopNotificationAllLabel',
        defaultMessage: 'All new messages',
    },
    mentions: {
        id: 'channel_notifications.desktopNotificationMentionLabel',
        defaultMessage: 'Mentions, direct messages, and keywords only',
    },
    nothing: {
        id: 'channel_notifications.desktopNotificationNothingLabel',
        defaultMessage: 'Nothing',
    },
});

const mobileNotificationInputFieldOptions = defineMessages({
    allNewMessages: {
        id: 'channel_notifications.MobileNotificationAllLabel',
        defaultMessage: 'All new messages',
    },
    mentions: {
        id: 'channel_notifications.MobileNotificationMentionLabel',
        defaultMessage: 'Mentions, direct messages, and keywords only',
    },
    nothing: {
        id: 'channel_notifications.MobileNotificationNothingLabel',
        defaultMessage: 'Nothing',
    },
});

const MuteChannelInputFieldData: FieldsetCheckbox = {
    title: translations.MuteChannelInputFieldTitle,
    name: 'mute channel',
    dataTestId: 'muteChannel',
};

const DesktopReplyThreadsInputFieldData: FieldsetCheckbox = {
    title: translations.DesktopReplyThreadsInputFieldTitle,
    name: 'desktop reply threads',
    dataTestId: 'desktopReplyThreads',
};

const MobileReplyThreadsInputFieldData: FieldsetCheckbox = {
    title: translations.MobileReplyThreadsInputFieldTitle,
    name: 'mobile reply threads',
    dataTestId: 'mobileReplyThreads',
};

const sameMobileSettingsDesktopInputFieldData: FieldsetCheckbox = {
    title: translations.sameMobileSettingsDesktopInputFieldTitle,
    name: 'same mobile settings as Desktop',
    dataTestId: 'sameMobileSettingsDesktop',
};

const IgnoreMentionsInputFieldData: FieldsetCheckbox = {
    title: translations.IgnoreMentionsInputFieldTitle,
    name: 'ignore mentions',
    dataTestId: 'ignoreMentions',
};

const AutoFollowThreadsInputFieldData: FieldsetCheckbox = {
    title: translations.AutoFollowThreadsInputFieldTitle,
    name: 'auto follow threads',
    dataTestId: 'autoFollowThreads',
};

const desktopNotificationInputFieldData = (defaultOption: string): FieldsetRadio => {
    return {
        options: [
            {
                dataTestId: `desktopNotification-${NotificationLevels.ALL}`,
                title: desktopNotificationInputFieldOptions.allNewMessages,
                name: `desktopNotification-${NotificationLevels.ALL}`,
                key: `desktopNotification-${NotificationLevels.ALL}`,
                value: NotificationLevels.ALL,
                suffix: defaultOption === NotificationLevels.ALL ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
            {
                dataTestId: `desktopNotification-${NotificationLevels.MENTION}`,
                title: desktopNotificationInputFieldOptions.mentions,
                name: `desktopNotification-${NotificationLevels.MENTION}`,
                key: `desktopNotification-${NotificationLevels.MENTION}`,
                value: NotificationLevels.MENTION,
                suffix: defaultOption === NotificationLevels.MENTION ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
            {
                dataTestId: `desktopNotification-${NotificationLevels.NONE}`,
                title: desktopNotificationInputFieldOptions.nothing,
                name: `desktopNotification-${NotificationLevels.NONE}`,
                key: `desktopNotification-${NotificationLevels.NONE}`,
                value: NotificationLevels.NONE,
                suffix: defaultOption === NotificationLevels.NONE ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
        ],
    };
};

const mobileNotificationInputFieldData = (defaultOption: string): FieldsetRadio => {
    return {
        options: [
            {
                dataTestId: `MobileNotification-${NotificationLevels.ALL}`,
                title: mobileNotificationInputFieldOptions.allNewMessages,
                name: `MobileNotification-${NotificationLevels.ALL}`,
                key: `MobileNotification-${NotificationLevels.ALL}`,
                value: NotificationLevels.ALL,
                suffix: defaultOption === NotificationLevels.ALL ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
            {
                dataTestId: `MobileNotification-${NotificationLevels.MENTION}`,
                title: mobileNotificationInputFieldOptions.mentions,
                name: `MobileNotification-${NotificationLevels.MENTION}`,
                key: `MobileNotification-${NotificationLevels.MENTION}`,
                value: NotificationLevels.MENTION,
                suffix: defaultOption === NotificationLevels.MENTION ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
            {
                dataTestId: `MobileNotification-${NotificationLevels.NONE}`,
                title: mobileNotificationInputFieldOptions.nothing,
                name: `MobileNotification-${NotificationLevels.NONE}`,
                key: `MobileNotification-${NotificationLevels.NONE}`,
                value: NotificationLevels.NONE,
                suffix: defaultOption === NotificationLevels.NONE ? (
                    <FormattedMessage
                        id='channel_notifications.default'
                        defaultMessage='(default)'
                    />) : undefined,
            },
        ],
    };
};

const utils = {
    desktopNotificationInputFieldData,
    DesktopNotificationsSectionDesc: translations.DesktopNotificationsSectionDesc,
    DesktopNotificationsSectionTitle: translations.DesktopNotificationsSectionTitle,
    IgnoreMentionsDesc: translations.IgnoreMentionsDesc,
    IgnoreMentionsInputFieldData,
    mobileNotificationInputFieldData,
    MobileNotificationsSectionDesc: translations.MobileNotificationsSectionDesc,
    MobileNotificationsSectionTitle: translations.MobileNotificationsSectionTitle,
    MuteAndIgnoreSectionTitle: translations.MuteAndIgnoreSectionTitle,
    MuteChannelDesc: translations.MuteChannelDesc,
    MuteChannelInputFieldData,
    NotifyMeTitle: translations.NotifyMeTitle,
    DesktopReplyThreadsInputFieldData,
    ThreadsReplyTitle: translations.ThreadsReplyTitle,
    MobileReplyThreadsInputFieldData,
    AutoFollowThreadsTitle: translations.AutoFollowThreadsTitle,
    AutoFollowThreadsDesc: translations.AutoFollowThreadsDesc,
    AutoFollowThreadsInputFieldData,
    sameMobileSettingsDesktopInputFieldData,
};

export default utils;
