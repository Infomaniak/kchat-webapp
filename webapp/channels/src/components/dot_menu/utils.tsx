// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type React from 'react';
import {EventTypes, TELEMETRY_CATEGORIES} from 'utils/constants';

import {trackEvent} from 'actions/telemetry_actions';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (e: ChangeEvent, suffix: string): void => {
    if (e.type === EventTypes.CLICK) {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.CLICK + '_' + suffix);
    } else {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.SHORTCUT + '_ ' + suffix);
    }
};
