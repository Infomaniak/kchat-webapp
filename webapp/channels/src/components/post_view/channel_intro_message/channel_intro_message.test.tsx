// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import type {Channel, ChannelType} from '@mattermost/types/channels';
import type {UserProfile} from '@mattermost/types/users';

import {Constants} from 'utils/constants';

import {renderWithContext, screen} from 'tests/react_testing_utils';

import ChannelIntroMessage from './channel_intro_message';

describe('components/post_view/ChannelIntroMessages', () => {
    const channel = {
        create_at: 1508265709607,
        creator_id: 'creator_id',
        delete_at: 0,
        display_name: 'test channel',
        header: 'test',
        id: 'channel_id',
        last_post_at: 1508265709635,
        name: 'testing',
        purpose: 'test',
        team_id: 'team-id',
        type: 'O',
        update_at: 1508265709607,
    } as Channel;

    const user1 = {id: 'user1', roles: 'system_user'};
    const users = [
        {id: 'user1', roles: 'system_user'},
        {id: 'guest1', roles: 'system_guest'},
    ] as UserProfile[];

    const baseProps = {
        currentUserId: 'test-user-id',
        channel,
        fullWidth: true,
        locale: 'en',
        channelProfiles: [],
        isReadOnly: false,
        isFavorite: false,
        enableUserCreation: false,
        teamIsGroupConstrained: false,
        creatorName: 'creatorName',
        currentUser: users[0],
        stats: {},
        usersLimit: 10,
        isMobileView: false,
        actions: {
            getTotalUsersStats: jest.fn().mockResolvedValue([]),
            favoriteChannel: jest.fn().mockResolvedValue([]),
            unfavoriteChannel: jest.fn().mockResolvedValue([]),
        },
    };

    describe('test Open Channel', () => {
        test('should match component state, without boards', () => {
            const initialState = {};
            const {container} = renderWithContext(
                <ChannelIntroMessage{...baseProps}/>, initialState,
            );

            const beginningHeading = screen.getByText('test channel');

            expect(beginningHeading).toBeInTheDocument();
            expect(beginningHeading).toHaveClass('channel-intro__title');

            expect(screen.getByText('This is the start of test channel. Any team member can join and read this channel.')).toBeInTheDocument();
        });
    });

    describe('test Group Channel', () => {
        const groupChannel = {
            ...channel,
            type: Constants.GM_CHANNEL as ChannelType,
        };
        const props = {
            ...baseProps,
            channel: groupChannel,
        };

        test('should match component state, no profiles', () => {
            const {container} = renderWithContext(
                <ChannelIntroMessage
                    {...props}
                />,
            );

            expect(screen.queryByText('test channel')).not.toBeInTheDocument();
            expect(screen.queryByText('Any member can join and read this channel.')).not.toBeInTheDocument();

            // there are no profiles in the dom, channel type is GM_CHANNEL, teammate text should be displayed
            expect(screen.getByText('This is the start of your group message history with these teammates. ', {exact: false})).toBeInTheDocument();

            expect(screen.getByText('This is the start of your', {exact: false})).toHaveClass('channel-intro__text');
        });

        test('should match component state, with profiles', () => {
            const {container} = renderWithContext(
                <ChannelIntroMessage
                    {...props}
                    channelProfiles={users}
                />,
            );

            expect(screen.getByText('This is the start of your group message history with these teammates. ', {exact: false})).toBeInTheDocument();

            const headerDialog = screen.getByLabelText('Set header dialog');
            expect(headerDialog).toBeInTheDocument();
            expect(headerDialog).toHaveTextContent('Set header');
            expect(headerDialog).toHaveClass('action-button');

            // one for user1 and one for guest

            const image = screen.getAllByAltText('user profile image');
            expect(image).toHaveLength(2);
            expect(image[0]).toHaveAttribute('src', '/api/v4/users/user1/image?_=0');
            expect(image[0]).toHaveAttribute('loading', 'lazy');

            expect(image[1]).toHaveAttribute('src', '/api/v4/users/guest1/image?_=0');
            expect(image[1]).toHaveAttribute('loading', 'lazy');

            const notificationPreferencesButton = screen.getByText('Notifications');
            expect(notificationPreferencesButton).toBeInTheDocument();
        });
    });

    describe('test DIRECT Channel', () => {
        const directChannel = {
            ...channel,
            type: Constants.DM_CHANNEL as ChannelType,
        };
        const props = {
            ...baseProps,
            channel: directChannel,
        };

        test('should match component state, without teammate', () => {
            const {container} = renderWithContext(
                <ChannelIntroMessage
                    {...props}
                />,
            );

            const message = screen.getByText('This is the start of your direct message history with this teammate. Messages and files shared here are not shown to anyone else.', {exact: false});
            expect(message).toBeInTheDocument();
            expect(message).toHaveClass('channel-intro__text');
        });

        test('should match component state, with teammate', () => {
            const {container} = renderWithContext(
                <ChannelIntroMessage
                    {...props}
                    teammate={user1 as UserProfile}
                    teammateName='my teammate'
                />,
            );
            expect(screen.getByText('This is the start of your direct message history with my teammate.', {exact: false})).toBeInTheDocument();

            const teammate = screen.getByText('my teammate');

            expect(teammate).toBeInTheDocument();
            expect(teammate).toHaveTextContent('my teammate');
            expect(teammate).toHaveClass('style--none');

            const image = screen.getByRole('img');

            expect(image).toBeInTheDocument();
            expect(image).toHaveAttribute('src', '/api/v4/users/user1/image?_=0');
            expect(image).toHaveAttribute('loading', 'lazy');

            const headerDialog = screen.getByLabelText('Set header dialog');

            expect(headerDialog).toBeInTheDocument();
            expect(headerDialog).toHaveTextContent('Set header');
            expect(headerDialog).toHaveClass('action-button');
        });
    });

    describe('test DEFAULT Channel', () => {
        const directChannel = {
            ...channel,
            name: Constants.DEFAULT_CHANNEL,
            type: Constants.OPEN_CHANNEL as ChannelType,
        };
        const props = {
            ...baseProps,
            channel: directChannel,
        };

        test('should match component state, readonly', () => {
            const {container} = renderWithContext(
                <ChannelIntroMessage
                    {...props}
                    isReadOnly={true}
                />,
            );

            const beginningHeading = screen.getByText('test channel');

            expect(beginningHeading).toBeInTheDocument();
            expect(beginningHeading).toHaveClass('channel-intro__title');

            expect(screen.getByText('Messages can only be posted by admins. Everyone automatically becomes a permanent member of this channel when they join the team.', {exact: false})).toBeInTheDocument();
        });

        test('should match component state without any permission', () => {
            const {container} = renderWithContext(
                <ChannelIntroMessage
                    {...props}
                    teamIsGroupConstrained={true}
                />,
            );
            expect(container).toMatchSnapshot();
        });

        test('should match snapshot, with enableUserCreation', () => {
            const wrapper = shallow(
                <ChannelIntroMessage
                    {...props}
                    enableUserCreation={true}
                />,
            );
            expect(wrapper).toMatchSnapshot();
        });

            const beginningHeading = screen.getByText('test channel');

            expect(beginningHeading).toBeInTheDocument();
            expect(beginningHeading).toHaveClass('channel-intro__title');
            expect(screen.getByText('Post messages here that you want everyone to see. Everyone automatically becomes a member of this channel when they join the team.', {exact: false})).toBeInTheDocument();
        });
    });

    describe('test OFFTOPIC Channel', () => {
        const directChannel = {
            ...channel,
            type: Constants.OPEN_CHANNEL as ChannelType,
            name: Constants.OFFTOPIC_CHANNEL,
        };
        const props = {
            ...baseProps,
            channel: directChannel,
        };

        test('should match component state', () => {
            const {container} = renderWithContext(
                <ChannelIntroMessage
                    {...props}
                />,
            );
            screen.getByText('This is the start of off-topic, a channel for non-work-related conversations.');
            expect(screen.getByText('This is the start of off-topic, a channel for non-work-related conversations.')).toHaveClass('channel-intro__text');
        });
    });
});
