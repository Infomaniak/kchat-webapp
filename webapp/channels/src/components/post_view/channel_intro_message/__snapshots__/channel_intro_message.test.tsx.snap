// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/ChannelIntroMessages test DEFAULT Channel should match snapshot, with enable, group constrained 1`] = `
<div
  className="channel-intro "
  id="channelIntro"
>
  <h2
    className="channel-intro__title"
  >
    <MemoizedFormattedMessage
      defaultMessage="Beginning of {name}"
      id="intro_messages.beginning"
      values={
        Object {
          "name": "test channel",
        }
      }
    />
  </h2>
  <p
    className="channel-intro__content"
  >
    <FormattedMarkdownMessage
      defaultMessage="**Welcome to {display_name}!**\\\\n \\\\nPost messages here that you want everyone to see. Everyone automatically becomes a permanent member of this channel when they join the team."
      id="intro_messages.default"
      values={
        Object {
          "display_name": "test channel",
        }
      }
    />
  </p>
  <Connect(TeamPermissionGate)
    permissions={
      Array [
        "invite_user",
      ]
    }
    teamId="team-id"
  >
    <Connect(TeamPermissionGate)
      permissions={
        Array [
          "add_user_to_team",
        ]
      }
      teamId="team-id"
    >
      <ToggleModalButton
        className="intro-links color--link"
        dialogProps={
          Object {
            "channel": Object {
              "create_at": 1508265709607,
              "creator_id": "creator_id",
              "delete_at": 0,
              "display_name": "test channel",
              "header": "test",
              "id": "channel_id",
              "last_post_at": 1508265709635,
              "name": "town-square",
              "purpose": "test",
              "team_id": "team-id",
              "type": "O",
              "update_at": 1508265709607,
            },
          }
        }
        dialogType={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        modalId="add_groups_to_team"
      >
        <i
          className="fa fa-user-plus"
        />
        <MemoizedFormattedMessage
          defaultMessage="Add other groups to this team"
          id="intro_messages.addGroupsToTeam"
        />
      </ToggleModalButton>
    </Connect(TeamPermissionGate)>
  </Connect(TeamPermissionGate)>
  <Connect(PluggableIntroButtons)
    channel={
      Object {
        "create_at": 1508265709607,
        "creator_id": "creator_id",
        "delete_at": 0,
        "display_name": "test channel",
        "header": "test",
        "id": "channel_id",
        "last_post_at": 1508265709635,
        "name": "town-square",
        "purpose": "test",
        "team_id": "team-id",
        "type": "O",
        "update_at": 1508265709607,
      }
    }
  />
  <Connect(Component)
    channelId="channel_id"
    permissions={
      Array [
        "manage_public_channel_properties",
      ]
    }
    teamId="team-id"
  >
    <ToggleModalButton
      ariaLabel="Set a Header"
      className="intro-links color--link channelIntroButton"
      dialogProps={
        Object {
          "channel": Object {
            "create_at": 1508265709607,
            "creator_id": "creator_id",
            "delete_at": 0,
            "display_name": "test channel",
            "header": "test",
            "id": "channel_id",
            "last_post_at": 1508265709635,
            "name": "town-square",
            "purpose": "test",
            "team_id": "team-id",
            "type": "O",
            "update_at": 1508265709607,
          },
        }
      }
      dialogType={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      modalId="edit_channel_header"
    >
      <EditIcon />
      <MemoizedFormattedMessage
        defaultMessage="Set a Header"
        id="intro_messages.setHeader"
      />
    </ToggleModalButton>
  </Connect(Component)>
  <br />
</div>
`;

exports[`components/post_view/ChannelIntroMessages test DEFAULT Channel should match snapshot, with enableUserCreation 1`] = `
<div
  className="channel-intro "
  id="channelIntro"
>
  <h2
    className="channel-intro__title"
  >
    <MemoizedFormattedMessage
      defaultMessage="Beginning of {name}"
      id="intro_messages.beginning"
      values={
        Object {
          "name": "test channel",
        }
      }
    />
  </h2>
  <p
    className="channel-intro__content"
  >
    <FormattedMarkdownMessage
      defaultMessage="**Welcome to {display_name}!**\\\\n \\\\nPost messages here that you want everyone to see. Everyone automatically becomes a permanent member of this channel when they join the team."
      id="intro_messages.default"
      values={
        Object {
          "display_name": "test channel",
        }
      }
    />
  </p>
  <Connect(TeamPermissionGate)
    permissions={
      Array [
        "invite_user",
      ]
    }
    teamId="team-id"
  >
    <Connect(TeamPermissionGate)
      permissions={
        Array [
          "add_user_to_team",
        ]
      }
      teamId="team-id"
    >
      <Memo(AddMembersButton)
        channel={
          Object {
            "create_at": 1508265709607,
            "creator_id": "creator_id",
            "delete_at": 0,
            "display_name": "test channel",
            "header": "test",
            "id": "channel_id",
            "last_post_at": 1508265709635,
            "name": "town-square",
            "purpose": "test",
            "team_id": "team-id",
            "type": "O",
            "update_at": 1508265709607,
          }
        }
        pluginButtons={
          <Memo(Connect(PluggableIntroButtons))
            channel={
              Object {
                "create_at": 1508265709607,
                "creator_id": "creator_id",
                "delete_at": 0,
                "display_name": "test channel",
                "header": "test",
                "id": "channel_id",
                "last_post_at": 1508265709635,
                "name": "town-square",
                "purpose": "test",
                "team_id": "team-id",
                "type": "O",
                "update_at": 1508265709607,
              }
            }
          />
        }
        setHeader={
          <Memo(Connect(Component))
            channelId="channel_id"
            permissions={
              Array [
                "manage_public_channel_properties",
              ]
            }
            teamId="team-id"
          >
            <ToggleModalButton
              ariaLabel="Set a Header"
              className="intro-links color--link channelIntroButton"
              dialogProps={
                Object {
                  "channel": Object {
                    "create_at": 1508265709607,
                    "creator_id": "creator_id",
                    "delete_at": 0,
                    "display_name": "test channel",
                    "header": "test",
                    "id": "channel_id",
                    "last_post_at": 1508265709635,
                    "name": "town-square",
                    "purpose": "test",
                    "team_id": "team-id",
                    "type": "O",
                    "update_at": 1508265709607,
                  },
                }
              }
              dialogType={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "type": [Function],
                }
              }
              modalId="edit_channel_header"
            >
              <EditIcon />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Set a Header"
                id="intro_messages.setHeader"
              />
            </ToggleModalButton>
          </Memo(Connect(Component))>
        }
        usersLimit={10}
      />
    </Connect(TeamPermissionGate)>
  </Connect(TeamPermissionGate)>
  <br />
</div>
`;
